src/main/java/core/basesyntax/handler/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.handler.impl;
+
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int currentQuantity = Storage.getFruitQuantity(transaction.getFruit());
+        if (currentQuantity < transaction.getQuantity()) {
+            throw new RuntimeException("Not enough stock for " + transaction.getFruit());
```

Comment:
When handling the Purchase operation, ensure that after the transaction, the resulting balance of the fruit in the Storage must not be negative. If it is, throw a RuntimeException with an informative message. The current implementation correctly throws an exception if the stock is insufficient, but the message could be more informative by including the attempted quantity and the current stock available.